### PR TITLE
fix: rename `estimated_bytes` to `memory_bytes` and expose `object_store_bytes` in ChunkSummary and system.chunks

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
closes https://github.com/influxdata/influxdb_iox/issues/1960

Improve observability of chunk resource usage (in preparation of starting to dig into query performance):

The current field name `estimated_bytes` can be confusing, especially as it is reported to be non-zero (but small) for parquet chunks whose "size", I for one, expect to reflect the size of the parquet file (not the in-memory statistics size)

I also want to be able to see the size of the parquet files we are working with

# Changes
1. Rename `estimated_bytes` to `memory_bytes`
2. add a new field `os_bytes` which represents the size in bytes on object store

Note the tracking of file size in parquet catalog was added in https://github.com/influxdata/influxdb_iox/pull/2006

### CODE REVIEW

File: server/src/db.rs [hunk 1]
```diff
@@ -2244,36 +2244,6 @@ mod tests {
         assert_eq!(read_buffer_chunk_ids(&db, partition_key), vec![1]);
     }
 
-    /// Normalizes a set of ChunkSummaries for comparison by removing timestamps
-    fn normalize_summaries(summaries: Vec<ChunkSummary>) -> Vec<ChunkSummary> {
```
Comment (alamb) [inline server/src/db.rs:2248]: This was a drive by refactoring (and I can put it in its own PR if people would prefer) -- it moves this code into `ChunkSummary` itself

File: server/src/db/catalog/chunk.rs [hunk 2]
```diff
@@ -550,6 +551,15 @@ impl CatalogChunk {
         }
     }
 
+    /// Returns the number of bytes of object storage consumed by this chunk
+    pub fn object_store_bytes(&self) -> usize {
+        match &self.stage {
+            ChunkStage::Open { .. } => 0,
+            ChunkStage::Frozen { .. } => 0,
+            ChunkStage::Persisted { parquet, .. } => parquet.file_size_bytes(),
```
Comment (alamb) [inline server/src/db/catalog/chunk.rs:559]: The core purpose of this PR is to make this value accessible -- lol

File: server/src/db.rs (hunk 1 above)
Comment (NGA-TRAN) [reply server/src/db.rs:2248]: no need to have it in its now PR.

File: server/src/db.rs [hunk 3]
```diff
@@ -2406,6 +2377,7 @@ mod tests {
                 ChunkStorage::ReadBufferAndObjectStore,
                 lifecycle_action,
                 2115, // size of RB and OS chunks
+                1132, // size of parquet file
```
Comment (NGA-TRAN) [inline server/src/db.rs:2380]: What 2115 includes? Size of RUB and size of the corresponding OS chunk that links to the parquet file whose size is 1132?

File: server/src/db/system_tables.rs [hunk 4]
```diff
@@ -640,15 +648,35 @@ mod tests {
...
+                time_of_first_write: Some(DateTime::from_utc(
+                    NaiveDateTime::from_timestamp(100, 0),
+                    Utc,
+                )),
+                time_of_last_write: Some(DateTime::from_utc(
+                    NaiveDateTime::from_timestamp(200, 0),
+                    Utc,
+                )),
+                time_closed: None,
+            },
```
Comment (NGA-TRAN) [inline server/src/db/system_tables.rs:669]: So this chunk is persisted and no longer in RUB but we still need 1234 bytes (or MB?) in memory to have information to access to the file of the chunk in OS?

**Review by NGA-TRAN — APPROVED** [review]
LGTM

File: server/src/db.rs (hunk 3 above)
Comment (alamb) [reply server/src/db.rs:2380]: The 2115  is the size of the data in RUB as well as the size of the in memory "ParquetChunk" structure (which is dominated by the size of the parquet statistics structure). It does not (confusingly to me) contain the size of the parquet data

File: server/src/db/system_tables.rs (hunk 4 above)
Comment (alamb) [reply server/src/db/system_tables.rs:669]: Yes -- primarily due to the size of the parquet statistics

File: server/src/db.rs (hunk 3 above)
Comment (NGA-TRAN) [reply server/src/db.rs:2398]: Sounds right to me

**Pull request merged**